### ISSUE TITLE
Replacer doesn't work for nested values

### ISSUE THREAD

**Issue by Akryum** [blakeembrey/javascript-stringify#27]
2.0.0 has a regression compared to 1.6.0. The replacer function doesn't seem to work on nested values, only on root-level ones.

[Reproduction](https://codesandbox.io/s/focused-saha-ov62e)

This example code doesn't produce the desired output:

```js
import { stringify } from "javascript-stringify";

function makeRaw(str) {
  const fn = () => {};
  fn.__expression = str;
  return fn;
}

function str(value) {
  return stringify(
    value,
    (val, indent, stringify) => {
      if (val && val.__expression) {
        return val.__expression;
      }
      return stringify(val);
    },
    2
  );
}

console.log(str({
  "no-console": makeRaw(`process.env.NODE_ENV === 'production' ? 'error' : 'off'`),
  "no-debugger": makeRaw(`process.env.NODE_ENV === 'production' ? 'error' : 'off'`)
}))
```

It outputs `() => {}` instead of `process.env.NODE_ENV === 'production' ? 'error' : 'off'` for example.

**Comment by Akryum** [comment]
cc @sodatea

**Comment by blakeembrey** [comment]
Thanks for the flag. It should be possible to fix this, it's unfortunately caused by a lot of new ways to write functions in ES2015 which resulted in having to write a custom function replacer that only worked with the object `key`.

**Comment by blakeembrey** [comment]
Fixed with https://github.com/blakeembrey/javascript-stringify/commit/6ce8383752d2e515d90895404dc38366f9cd50a5 and released as `2.0.1`.
